x-pack/legacy/plugins/ml/public/jobs/new_job_new/common/job_creator/util/general.ts

```diff
@@ -51,6 +57,64 @@ function getDetectors(job: Job, datafeed: Datafeed) {
         },
       ];
     }
+  } else {
+    // all other detectors.
+    detectors = detectors.map(d => {
+      // if a count function is used, add EVENT_RATE_FIELD_ID as its field
+      if (d.function === ML_JOB_AGGREGATION.COUNT) {
+        return { ...d, field_name: EVENT_RATE_FIELD_ID };
+      }
+      if (d.function === ML_JOB_AGGREGATION.HIGH_COUNT) {
+        return { ...d, field_name: EVENT_RATE_FIELD_ID };
+      }
+      if (d.function === ML_JOB_AGGREGATION.LOW_COUNT) {
+        return { ...d, field_name: EVENT_RATE_FIELD_ID };
+      }
+      // if sparse data functions were used, replace them with their non-sparse versions
+      // the sparse data flag has already been determined and set, so this information is not being lost.
+      if (d.function === ML_JOB_AGGREGATION.NON_ZERO_COUNT) {
+        return { ...d, field_name: EVENT_RATE_FIELD_ID, function: ML_JOB_AGGREGATION.COUNT };
+      }
+      if (d.function === ML_JOB_AGGREGATION.HIGH_NON_ZERO_COUNT) {
+        return { ...d, field_name: EVENT_RATE_FIELD_ID, function: ML_JOB_AGGREGATION.HIGH_COUNT };
+      }
+      if (d.function === ML_JOB_AGGREGATION.LOW_NON_ZERO_COUNT) {
+        return { ...d, field_name: EVENT_RATE_FIELD_ID, function: ML_JOB_AGGREGATION.LOW_COUNT };
+      }
+      if (d.function === ML_JOB_AGGREGATION.NON_NULL_SUM) {
+        return { ...d, function: ML_JOB_AGGREGATION.SUM };
+      }
+      if (d.function === ML_JOB_AGGREGATION.HIGH_NON_NULL_SUM) {
+        return { ...d, function: ML_JOB_AGGREGATION.HIGH_SUM };
+      }
+      if (d.function === ML_JOB_AGGREGATION.LOW_NON_NULL_SUM) {
+        return { ...d, function: ML_JOB_AGGREGATION.LOW_SUM };
+      }
+      return d;
+    });
   }
   return detectors;
 }
+
+// determine whether the job has been configured to run on sparse data
+// by looking to see whether the datafeed contains a dc_region field in an aggregation
+// if it does, it is a distinct count single metric job and no a sparse data job.
+// this check is needed because distinct count jobs also use NON_ZERO_COUNT
+export function determineSparseDataJob(job: Job, datafeed: Datafeed) {
```

Comment:
nit: maybe `isSparseDataJob()` would make it more obvious that this returns a `boolean`.